docs_src/design/adr/0024-system-events.md

```diff
@@ -0,0 +1,93 @@
+
+The new `SystemEvent` DTO will be published to a multi-level topic allowing subscribers to filter by topic. The format of this topic for Device System Events will be:
+
+​		`edgex/system-event/{event-type}/{event-action}/{event-source}/{event-owner}/{event-other}`
+
+where 
+
+- `{event-type}` = device
+- `{event-action}` = add, update or delete
+- `{event-source}` = core-metadata
+- `{event-owner}` = `<device service name>`
+- `{event-other}` = `<device profile name>`
+
+As noted above Event Owner and Event Other are optional based on the Event Type and may not be meaningful in some future use cases. Those use cases will omit the Event Owner and Event Other from the publish topic as appropriate
+
+This multi-level topic scheme allows consuming services to filter which System Events they receive base of the topic that is subscribed.
+
+!!! example - "Example - System Event subscription topics"
+    ```
+    edgex/system-event/# - All system events
+    edgex/system-event/device/# - only device system events
+    edgex/system-event/#/#/core-metadata/# - only system events from Core Metadata
+    edgex/system-event/device/add/#/device-onvif-camera/# - only add device system events for device-onvif-camera
+    edgex/system-event/device/#/#/#/onvif-camera - only device system events for devices created for the onvif-camera device profile
+    ```
+
+#### Consumers
+
+Consumers of Device System Events will likely be custom applications service as described in [System Events for Devices ](https://docs.edgexfoundry.org/2.3/design/ucr/0001-System-Events-for-Devices/). No changes are required to the App Functions SDK since it already supports processing of different types via the [Target Type](https://docs.edgexfoundry.org/2.2/microservices/application/AdvancedTopics/#target-type) capability. Developers of custom App Services that consume System Events will need to do the following:
+
+- Set the Target Type to `&dtos.SystemEvent{}`when creating instance of `ApplicationService` using the [NewAppServiceWithTargetType](https://docs.edgexfoundry.org/2.2/microservices/application/ApplicationServiceAPI/#newappservicewithtargettype) factory function.
+- Write custom pipeline function that expects the new `SystemEvent` DTO and process it accordingly.  Similar to how the [ToLineProtocol](https://docs.edgexfoundry.org/2.2/microservices/application/BuiltIn/#tolineprotocol) pipeline function expects the Metric DTO.
+
+### Services/Modules impacted
+
+- Core Metadata service
+    - This service is the single point for device Add/Update/Delete and will be the producer of Device System Events.
+- Core Contracts module
+    - The new `SystemEvent` DTO will be added to this repository
+- Camera Management App Service Example
+    - Once Device System Event are implemented, the [Camera Management](https://github.com/edgexfoundry/edgex-examples/tree/main/application-services/custom/camera-management) example can be updated to consume the new Device System Events.
```

Comment:
Add Device Service to the impacted list, because this approach should replace the normal REST callback in the future

Reply:
Excellent point! Will add.